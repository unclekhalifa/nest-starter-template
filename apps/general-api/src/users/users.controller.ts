import {
  Controller,
  Post,
  Body,
} from '@nestjs/common';
import { UsersService } from './users.service';
import { CreateUserDto } from './dto/create-user.dto';
import { AdminAuth } from '@nest-workspace/auth';

@Controller('users')
export class UsersController {
  constructor(private readonly usersService: UsersService) {}

  @AdminAuth()
  @Post()
  create(@Body() createUserDto: CreateUserDto) {
    return this.usersService.create(createUserDto);
  }
}
