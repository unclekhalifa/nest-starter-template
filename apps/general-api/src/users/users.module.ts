import { Module } from '@nestjs/common';
import { UsersService } from './users.service';
import { UsersController } from './users.controller';
import { DatabaseModule } from '../database';

@Module({
  imports: [
    DatabaseModule,
  ],
  controllers: [UsersController],
  providers: [UsersService],
})
export class UsersModule {}
