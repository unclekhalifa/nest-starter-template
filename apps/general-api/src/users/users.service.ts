import { Injectable } from '@nestjs/common';
import { CreateUserDto } from './dto/create-user.dto';
import { UpdateUserDto } from './dto/update-user.dto';
import { DatabaseService } from '../database';

@Injectable()
export class UsersService {
  constructor(private readonly database: DatabaseService) {}

  create(createUserDto: CreateUserDto) {
    return this.database.user.create({ data: createUserDto });
  }

  findAll() {
    return `This action returns all users`;
  }

  findOne(id: number) {
    return `This action returns a #${id} user`;
  }

  update(id: number, updateUserDto: UpdateUserDto) {
    return `This action updates a #${id} user`;
  }

  remove(id: number) {
    return `This action removes a #${id} user`;
  }
}
