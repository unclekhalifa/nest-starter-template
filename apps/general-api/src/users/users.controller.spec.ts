import { Test, TestingModule } from '@nestjs/testing';
import { UsersController } from './users.controller';
import { UsersService } from './users.service';
import { DatabaseService } from '../database';

describe('UsersController', () => {
  let controller: UsersController;

  beforeEach(async () => {
    const module: TestingModule = await Test.createTestingModule({
      controllers: [UsersController],
      providers: [UsersService, DatabaseService],
    }).compile();

    controller = module.get<UsersController>(UsersController);
  });

  it('should be defined', () => {
    expect(controller).toBeDefined();
  });

  describe('create', () => {
    it('should call create on the users service', () => {
      const createUserDto = { name: 'John Doe', email: 'john.doe@test.com' };
      controller.create(createUserDto);
      expect(createUserDto).toMatchSnapshot();
    });
  });
});
