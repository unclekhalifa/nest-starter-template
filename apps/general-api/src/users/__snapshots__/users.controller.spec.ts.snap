// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersController create should call create on the users service 1`] = `
{
  "email": "john.doe@test.com",
  "name": "John Doe",
}
`;
